File: aoc_1.R
count_increases <- function(x, n) {
  ln <- length(x) + 1
  x <- rowSums(matrix(x, ln, n)[-ln, ] |>
    suppressWarnings() |> as.matrix())
  sum(diff(x) > 0)
}

d <- as.integer(readLines("data/aoc_1"))
count_increases(d, 1)
count_increases(d, 3)

File: aoc_1.jl
function count_increases(x)
    sum(diff(x) .> 0)
end

function sum_window(x, n)
    n == 1 && return count_increases(x)

    ln = length(x)
    out = zeros(Int, ln)
    for i in 1:ln - n + 1
        out[i] = sum(x[i:(i + n - 1)])
    end
    return count_increases(out)
end

data = parse.(Int, readlines("data/aoc_1"))
sum_window(data, 1)
sum_window(data, 3)
